### PR TITLE
Cy 3413 support local directory part 2

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: cloudify_tf/utils.py [hunk 1]
```diff
@@ -120,119 +190,429 @@ def clean_strings(string):
...
+                plugins=plugins_dir, storage=storage_path))
+    ctx.logger.debug('Value plugins_dir is {loc}.'.format(
+        loc=plugins_dir))
+    return plugins_dir
+
+
+def get_plugins(target=False):
+    """These are plugins that the user wishes to install."""
+    resource_config = get_resource_config(target=target)
+    return resource_config.get('plugins', [])
```
Comment (AdarShaked) [inline cloudify_tf/utils.py:422]: {} ?

File: cloudify_tf/utils.py [hunk 2]
```diff
@@ -120,119 +190,429 @@ def clean_strings(string):
...
+    and then store it again for later use.
+    """
+    module_root = get_storage_path()
+    source_path = get_source_path()
+    extract_binary_tf_data(module_root, material, source_path)
+    handle_backend(module_root)
     try:
-        yield module_root
+        yield get_node_instance_dir()
     finally:
```
Comment (AdarShaked) [inline cloudify_tf/utils.py:544]: why you need to zip it in finally?

File: plugin.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ plugins:
   tf:
     executor: central_deployment_agent
     package_name: cloudify-terraform-plugin
-    package_version: '0.13.4'
```
Comment (AdarShaked) [inline plugin.yaml:5]: not 0.14.0 ? its a feature.

File: cloudify_tf/tasks.py [hunk 4]
```diff
@@ -115,187 +99,114 @@ def reload_template(ctx, source, destroy_previous, **_):
...
-        if os.path.isdir(dir_to_delete):
-            ctx.logger.info("Removing %s: %s", property_desc, dir_to_delete)
-
-            shutil.rmtree(dir_to_delete)
-        else:
-            ctx.logger.info("Directory %s doesn't exist; skipping",
-                            dir_to_delete)
+            ('plugins_dir', 'plugins directory'),
+            ('storage_path', 'storage_directory')]:
+        dir_to_delete = terraform_config.get('{0}'.format(property_name), '')
```
[the remark this answers is not in the document]
Comment (EarthmanT) [reply cloudify_tf/tasks.py:167]: seems left over from last time

File: cloudify_tf/utils.py (hunk 2 above)
Comment (EarthmanT) [reply cloudify_tf/utils.py:544]: The source is stored as base64 encoded zip. That's just the way the plugin is written.

File: setup.py [hunk 5]
```diff
@@ -28,7 +47,7 @@
         "cloudify-common>=4.5.5",
         "pyyaml==3.12",
         "requests>=2.7.0,<3.0",
-        "cloudify-utilities-plugins-sdk==0.0.21",  # Shared Resource Downloader
+        "cloudify-utilities-plugins-sdk>=0.0.30",  # Shared Resource Downloader
```
Comment (AdarShaked) [inline setup.py:50]: 0.0.31 for get_deployment_dir

File: cloudify_tf/tests/test_plugin.py [hunk 6]
```diff
@@ -12,17 +12,40 @@
...
+from ..tasks import (install,
+                     set_directory_config)
+from ..utils import RELATIONSHIP_INSTANCE
+
+
+test_dir1 = mkdtemp()
+test_dir2 = mkdtemp()
+
+
+def get_dir1(*foo, **bar):
```
Comment (AdarShaked) [inline cloudify_tf/tests/test_plugin.py:36]: Delete.

File: cloudify_tf/tests/test_plugin.py [hunk 7]
```diff
@@ -12,17 +12,40 @@
...
+    return test_dir1
+
+
+def get_dir2(*foo, **bar):
```
Comment (AdarShaked) [inline cloudify_tf/tests/test_plugin.py:40]: Delete.

**Review by AdarShaked — CHANGES_REQUESTED** [review]
Must  update utilities sdk vresion.

**Review by AdarShaked — APPROVED** [review]

**Pull request merged**